hw_6/script.js

```diff
@@ -0,0 +1,520 @@
+                        parent: "div",
+                    },
+                ]
+            }
+        ]
+    }
+    console.log(body);
+}
+// Destructure
+{
+    const body = {
+        tagName: "body",
+        children: [
+          {
+            tagName: "div",
+            children: [
+              {
+                tagName: "span",
+                children: ["Enter a data please:"],
+              },
+              {
+                tagName: "br",
+              },
+              {
+                tagName: "input",
+                attr: {
+                  type: "text",
+                  id: "name",
+                },
+              },
+              {
+                tagName: "input",
+                attr: {
+                  type: "text",
+                  id: "surname",
+                },
+              },
+            ],
+          },
+          {
+            tagName: "div",
+            children: [
+              { 
+                tagName: "button", 
+                attr: { id: "ok" }, 
+                children: ["Ok"] 
+            },
+              { 
+                tagName: "button", 
+                attr: { id: "cancel" }, 
+                children: ["Cancel"] 
+            },
+            ],
+          },
+        ],
+    };
+    const {
+        children: [
+          {
+            children: [
+              {
+                children: [text],
+              },
+            ],
+          },
+        ],
+      } = body;
+      console.log("span:", text);
+    
+      const {
+        children: [
+          ,
+          {
+            children: [
+              ,
+              {
+                children: [btnText],
+              },
+            ],
+          },
+        ],
+      } = body;
+      console.log("children:", btnText);
+
+      const {
+        children: [
+          {
+            children: [
+              ,
+              ,
+              ,
+              {
+                attr: { id },
+              },
+            ],
+          },
+        ],
+      } = body;
+      console.log("input:", id);
+}
+// Destruct array
+{
+    let arr = [1,2,3,4,5, "a", "b", "c"];
+    let [odd1, even1, odd2, even2, odd3, ...letters] = arr;
+    console.log("Odd Numbers:", odd1, odd2, odd3);
+    console.log("Even Numbers:", even1, even2);
+    console.log("Letters:", letters);
+}
+// Destruct string
+{
+    let arr = [1, "abc"];
+    let [number, str] = arr;
+    let [s1, s2, s3] = str.split('');
+    console.log("Number:", number);
+    console.log("s1:", s1);
+    console.log("s2:", s2);
+    console.log("s3:", s3);
+}
+// Destruct 2
+{
+    const obj = {name: 'Ivan',
+    surname: 'Petrov',
+    children: [{name: 'Maria'}, {name: 'Nikolay'}]}
+
+    const { children: [{ name: name1 }, { name: name2 }] } = obj;
+
+    console.log("Name of child 1:", name1);
+    console.log("Name of child 2:", name2);
+}
+// Destruct 3
+{
+    let arr = [1, 2, 3, 4, 5, 6, 7, 10];
+
+    let [a, b, ...rest] = arr;
```

Comment:
use {} for arr
arr is a secret object
`     const { 0: a, 1: b, length } = arr;`